mlir/lib/Linker/LinkerInterface.cpp

```diff
@@ -130,7 +130,8 @@ LogicalResult SymbolAttrLinkerInterface::link(LinkState &state) const {
     if (!materialized)
       return materializeError(op);
 
-    st.insert(materialized);
+    if (isa<SymbolOpInterface>(op))
+      st.insert(materialized);
```

Comment:
I don't exactly like this solution, since it's not generally relevant to `SymbolAttrLinkerInterface` but I'm also not sure how to do this cleanly and without copy-pasting most of this function